modules/instances/main.tf

```diff
@@ -90,7 +90,7 @@ resource "azurerm_linux_virtual_machine" "server" {
   # Due to the nature of azure resources there is no single resource which presents in terraform both public IP and internal DNS
   # for consistency with other providers I thought it would work best to put this tag on the instance
   tags        = merge({
-    internalDNS = "pe-server-${count.index}-${var.id}.${azurerm_network_interface.server_nic[count.index].internal_domain_name_suffix}"
+    internalDNS = var.domain_name == null ? "pe-server-${count.index}-${var.id}.${azurerm_network_interface.server_nic[count.index].internal_domain_name_suffix}" : "pe-server-${count.index}-${var.id}.${var.domain_name}"
```

Comment:
Can we do the same refactor trick here to prevent code duplication?

Reply:
In this context the name is constructed using `count.index` which is a property of the resource so the construction must occur within the the resource. I have vague idea of a work around but it should be applied to all three modules if it worked so more appropriate for another PR